kythe/proto/repo.proto

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2018 The Kythe Authors. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+syntax = "proto3";
+
+package kythe.proto.repo;
+
+// Represents configuration data on how to extract a repo.
+message Config {
+  repeated ExtractionHint extractions = 1;
```

Comment:
Is this repeated because there could be multiple ways to extract the same source, because there are multiple ExtractionHints needed for one extraction for the same source, or because there could be multiple source roots in the same repo?

Reply:
Sort of in between?  In particular, I want some flexibility on how to handle one repo that has either multiple source roots (e.g. separate maven sub parts), or even multiple source roots with different builders (I've seen maven+gradle, but I also imagine backend+frontend will be a common theme in the future).

On the other hand, I could be easily convinced to make this not repeated for the short term (in fact the code I have ready and waiting explicitly only supports size=1 right now).